File: xiang.li/1200.kt
import kotlin.math.abs

class Solution {
    fun minimumAbsDifference(arr: IntArray): List<List<Int>> {
        val len = arr.size
        arr.sort()
        val c = arr.mapIndexed { index, i -> Pair(i, arr[(index + 1)  % len])}.sortedBy { abs(it.first - it.second) }
        return c.filter { abs(it.first - it.second) == abs(c[0].first - c[0].second) }.map { it.toList() }
    }
}

File: hang.liu/1200.js
var minimumAbsDifference = function(arr) {
  arr.sort((a, b) => a - b);

  var result = Array(arr.length - 1);

  var min = arr[1] - arr[0];

  for (var i = 0; i < arr.length - 1; i++) {
    var diff = arr[i + 1] - arr[i];
    result[i] = [diff, arr[i], arr[i + 1]];
    if (diff < min) min = diff;
  }

  return result.filter(item => item[0] === min).map(item => [item[1], item[2]]);
};
